modules/gcp/storage.go

```diff
@@ -0,0 +1,214 @@
+func ReadBucketObject(t *testing.T, bucketName string, filePath string) io.Reader {
+	out, err := ReadBucketObjectE(t, bucketName, filePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return out
+}
+
+// ReadBucketObjectE reads an object from the given Storage Bucket and returns its contents.
+func ReadBucketObjectE(t *testing.T, bucketName string, filePath string) (io.Reader, error) {
+	logger.Logf(t, "Reading object from bucket %s using path %s", bucketName, filePath)
+
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	bucket := client.Bucket(bucketName)
+	r, err := bucket.Object(filePath).NewReader(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// WriteBucketObject writes an object to the given Storage Bucket and returns its URL.
+func WriteBucketObject(t *testing.T, bucketName string, filePath string, body io.Reader, contentType string) string {
+	out, err := WriteBucketObjectE(t, bucketName, filePath, body, contentType)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return out
+}
+
+// WriteBucketObjectE writes an object to the given Storage Bucket and returns its URL.
+func WriteBucketObjectE(t *testing.T, bucketName string, filePath string, body io.Reader, contentType string) (string, error) {
+	// set a default content type
+	if contentType == "" {
+		contentType = "application/octet-stream"
+	}
+
+	logger.Logf(t, "Writing object to bucket %s using path %s and content type %s", bucketName, filePath, contentType)
+
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	w := client.Bucket(bucketName).Object(filePath).NewWriter(ctx)
+	w.ContentType = contentType
+
+	// Don't set any ACL or cache control properties for now
+	//w.ACL = []storage.ACLRule{{Entity: storage.AllAuthenticatedUsers, Role: storage.RoleReader}}
+	// set a default cache control (1 day)
+	//w.CacheControl = "public, max-age=86400"
+
+	if _, err := io.Copy(w, body); err != nil {
+		return "", err
+	}
+	if err := w.Close(); err != nil {
+		return "", err
+	}
+
+	const publicURL = "https://storage.googleapis.com/%s/%s"
+	return fmt.Sprintf(publicURL, bucketName, filePath), nil
+}
+
+// EmptyStorageBucket removes the contents of a storage bucket with the given name.
+func EmptyStorageBucket(t *testing.T, name string) {
+	err := EmptyStorageBucketE(t, name)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// EmptyStorageBucketE removes the contents of a storage bucket with the given name.
+func EmptyStorageBucketE(t *testing.T, name string) error {
+	logger.Logf(t, "Emptying storage bucket %s", name)
+
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// List all objects in the bucket
+	//
+	// TODO - we should really do a bulk delete call here, but I couldn't find
+	// anything in the SDK.
```

Comment:
AFAIK, the AWS API has the same limitations!

Reply:
Dang!